model/converter/thrift/jaeger/from_domain_test.go

```diff
@@ -0,0 +1,147 @@
+package jaeger
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/uber/jaeger/model"
+	j "github.com/uber/jaeger/thrift-gen/jaeger"
+)
+
+func spanRefsEqual(refs []*j.SpanRef, otherRefs []*j.SpanRef) bool {
+	if len(refs) != len(otherRefs) {
+		return false
+	}
+
+	for idx, ref := range refs {
+		if *ref != *otherRefs[idx] {
+			return false
+		}
+	}
+	return true
+}
+
+type spanOptions struct {
+	TraceIDLow    uint64
+	TraceIDHigh   uint64
+	SpanID        uint64
+	ParentSpanID  uint64
+	OperationName string
+	References    []model.SpanRef
+	Flags         int32
+	StartTime     time.Time
+	Duration      time.Duration
+	Tags          model.KeyValues
+	Logs          []model.Log
+	Process       model.Process
+}
+
+func generateRandomSpan(options *spanOptions) *model.Span {
```

Comment:
I think we have a flyer in the restrooms that says "don't put logic in tests". Why all these complications?

The best way I would test this converter is to go through all the fixtures in model/converter/thrift/jaeger/fixtures and do the following:
  1. load the domain model JSON into model.Trace
  2. convert to jaeger.thrift spans using your new converter
  3. convert back to domain model using the other converter in this package
  4. do a deep compare of the domain objects (assert.Equals should do the trick)